### ISSUE TITLE
Exception with GraphQL extended type

### ISSUE THREAD

**Issue by hjaekel** [quarkusio/quarkus#19879]
### Describe the bug

When an extended GraphQL type is defined in the resource class, the extended type for the first result element cannot be loaded, but the extended types for the next result elements are loaded correctly.

Here is the example GraphQL query, where the movie type is extended with the actor type:

```
query getMovieAndActor{allMovies {
  id title actors{id name}
}}
```

### Expected behavior

The expected outcome is that all actors for all movies are loaded:

```json
{
  "data": {
    "allMovies": [
      {
        "id": 1,
        "title": "Reservoir Dogs",
        "actors": [
          {
            "id": 2,
            "name": "John Travolta"
          }
        ]
      },
      {
        "id": 2,
        "title": "Pulp Fiction",
        "actors": [
          {
            "id": 1,
            "name": "Quentin Tarantino"
          },
          {
            "id": 2,
            "name": "John Travolta"
          },
          {
            "id": 3,
            "name": "Samuel L Jackson"
          }
        ]
      },
      {
        "id": 3,
        "title": "The Mask",
        "actors": [
          {
            "id": 4,
            "name": "Jim Carrey"
          }
        ]
      }
    ]
  }
}
```

### Actual behavior

Exception:

```stacktrace
2021-09-02 19:45:11,821 ERROR [io.sma.graphql] (executor-thread-0) SRGQL012000: Data Fetching Error: java.lang.IllegalStateException: HR000068: This method should exclusively be invoked from a Vert.x EventLoop thread; currently running on thread 'executor-thread-0'
	at org.hibernate.reactive.common.InternalStateAssertions.assertUseOnEventLoop(InternalStateAssertions.java:40)
	at org.hibernate.reactive.session.impl.ReactiveSessionImpl.getReactiveConnection(ReactiveSessionImpl.java:1481)
	at org.hibernate.reactive.mutiny.impl.MutinySessionImpl$Transaction.begin(MutinySessionImpl.java:454)
	at org.hibernate.reactive.mutiny.impl.MutinySessionImpl$Transaction.execute(MutinySessionImpl.java:433)
	at org.hibernate.reactive.mutiny.impl.MutinySessionImpl.withTransaction(MutinySessionImpl.java:417)
	at io.quarkus.hibernate.reactive.runtime.ReactiveSessionProducer_ProducerMethod_createMutinySession_1321d110ee9e92bda147899150401e0a136779c7_ClientProxy.withTransaction(ReactiveSessionProducer_ProducerMethod_createMutinySession_1321d110ee9e92bda147899150401e0a136779c7_ClientProxy.zig:615)
	at io.quarkus.hibernate.reactive.panache.Panache.lambda$withTransaction$1(Panache.java:39)
	at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.performInnerSubscription(UniOnItemTransformToUni.java:68)
	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni$UniOnItemTransformToUniProcessor.onItem(UniOnItemTransformToUni.java:57)
	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem$KnownItemSubscription.forward(UniCreateFromKnownItem.java:38)
	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem$KnownItemSubscription.access$100(UniCreateFromKnownItem.java:26)
	at io.smallrye.mutiny.operators.uni.builders.UniCreateFromKnownItem.subscribe(UniCreateFromKnownItem.java:23)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
	at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureFlatMap.subscribe(UniOnItemOrFailureFlatMap.java:27)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:36)
	at io.smallrye.mutiny.operators.uni.UniRunSubscribeOn.lambda$subscribe$0(UniRunSubscribeOn.java:27)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:536)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

GraphQL result:

```json
{
  "errors": [
    {
      "message": "Server Error",
      "locations": [
        {
          "line": 2,
          "column": 12
        }
      ],
      "path": [
        "allMovies",
        0,
        "actors"
      ],
      "extensions": {
        "classification": "DataFetchingException"
      }
    }
  ],
  "data": {
    "allMovies": [
      {
        "id": 1,
        "title": "Reservoir Dogs",
        "actors": null
      },
      {
        "id": 2,
        "title": "Pulp Fiction",
        "actors": [
          {
            "id": 1,
            "name": "Quentin Tarantino"
          },
          {
            "id": 2,
            "name": "John Travolta"
          },
          {
            "id": 3,
            "name": "Samuel L Jackson"
          }
        ]
      },
      {
        "id": 3,
        "title": "The Mask",
        "actors": [
          {
            "id": 4,
            "name": "Jim Carrey"
          }
        ]
      }
    ]
  }
}
```

### How to Reproduce?

1. $ git clone https://github.com/gaf-ag/quarkus-reactive-hibernate-graphql-crud
2. $ cd quarkus-reactive-hibernate-graphql-crud 
2. $ git checkout 2.1.4.Final-extended-GraphQL-type-bug
3. $ docker run -d --rm --name my_reative_db -e POSTGRES_USER=user -e POSTGRES_PASSWORD=password -e POSTGRES_DB=my_db -p 5432:5432 postgres:10.5
4. $ ./mvnw compile quarkus:dev
5. Go to http://localhost:8080/q/graphql-ui/
6. Execute the following GraphQL query:
```
query getMovieAndActor{allMovies {
  id title actors{id name}
}}
```

### Output of `uname -a` or `ver`

Darwin xxx 20.6.0 Darwin Kernel Version 20.6.0: Wed Jun 23 00:26:31 PDT 2021; root:xnu-7195.141.2~5/RELEASE_X86_64 x86_64

### Output of `java -version`

OpenJDK 64-Bit Server VM Temurin-11.0.12+7 (build 11.0.12+7, mixed mode)

### GraalVM version (if different from Java)

n/a

### Quarkus version or git rev

2.1.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)

**Comment by hjaekel** [comment]
Hi @jmartisk, @phillip-kruger, could you already look at this issue? Can I somehow support your investigation for this issue?

I tried my example with the current Quarkus 2.2.3.Final. The result is the same (the extended type for the first result element cannot be loaded). What changed is that the `DataFetchingException` is not logged on the server side.

**Comment by phillip-kruger** [comment]
Looking at it now.

**Comment by phillip-kruger** [comment]
This is tied to https://github.com/smallrye/smallrye-graphql/issues/918 and on my TODO list to fix a.s.a.p.

**Comment by hjaekel** [comment]
That sounds great, thank you Philip.

**Comment by phillip-kruger** [comment]
This should be fixed now in v 2.10.0. Please reopen if not.
